torch/nn/modules/normalization.py

```diff
@@ -154,8 +154,8 @@ def __init__(self, normalized_shape: _shape_t, eps: float = 1e-5, elementwise_af
         self.eps = eps
         self.elementwise_affine = elementwise_affine
         if self.elementwise_affine:
-            self.weight = Parameter(torch.Tensor(*self.normalized_shape))
-            self.bias = Parameter(torch.Tensor(*self.normalized_shape))
+            self.weight = Parameter(torch.empty(*(self.normalized_shape or (0,))))
```

Comment:
Why is the `or (0,)` part necessary here and below?

Reply:
Otherwise the test in _test/test_nn.py_ would fail:

https://github.com/pytorch/pytorch/blob/527c1e0e37b7c65148bcbc390b65e94fb4624a9d/test/test_nn.py#L11082-L11092

`torch.empty` with `()` as input (the first "bad input" case) raises an error:

```python
TypeError: empty() received an invalid combination of arguments - got (), but expected one of:
 * (tuple of ints size, *, tuple of names names, torch.memory_format memory_format, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
 * (tuple of ints size, *, torch.memory_format memory_format, Tensor out, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
```

Another option would be to remove that first "bad input" case (`(): ()`).

Reply:
I think the problem comes from doing arg expansion in the call to `torch.empty()`:

```python
>>> torch.empty(())
tensor(0.)
>>> torch.empty(*())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: empty() received an invalid combination of arguments - got (), but expected one of:
 * (tuple of ints size, *, tuple of names names, torch.memory_format memory_format, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
 * (tuple of ints size, *, torch.memory_format memory_format, Tensor out, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
```

This can be fixed by passing `self.normalized_shape` directly to `torch.empty()` without the `*`.

Reply:
Ah, you are right! That does sound like a better option.